feature-libs/pickup-in-store/cart/facade/pickup-in-store-active-cart.service.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import {
+  ActiveCartService,
+  getCartIdByUserId,
+} from '@spartacus/cart/base/core';
+import { MultiCartFacade } from '@spartacus/cart/base/root';
+import { UserIdService } from '@spartacus/core';
+import { IntendedPickupLocationFacade } from '@spartacus/pickup-in-store/root';
+import { withLatestFrom } from 'rxjs/operators';
+
+/**
+ * An override of the active cart service that can add a product for pickup in store.
+ */
+@Injectable()
+export class PickupInStoreActiveCartService extends ActiveCartService {
+  constructor(
+    protected userIdService: UserIdService,
+    protected multiCartFacade: MultiCartFacade,
+    protected intendedPickupLocationFacade: IntendedPickupLocationFacade
+  ) {
+    super(multiCartFacade, userIdService);
+  }
+
+  addEntry(
```

Comment:
No need to replace our `addEntry` function. This function is used in many places, and pickupInStore is not always the intended store.
go to `add-to-cart.component.ts`, at line 155, you can do something like this to pass store to "add-to-cart"
```
this.pickupOptionCompRef.instance.displayPickupLocation$.pipe(take(1)).subscribe(
        (value: string | undefined) => {
          console.log('pickup store: ', value);
          this.pickupStore = value;
        }
      );
```

Reply:
Done